### ISSUE TITLE
Wrong Application Name

### ISSUE THREAD

**Issue by sedrubal** [safing/portmaster#747]
### What happened?
When portmaster shows a notification that an application is trying to access the internet, sometimes the wrong application name is shown. As well, the wrong rules are applide. For example, I&#39;m accessing a website using Firefox (which is allowed to access everything) but portmaster thinks my mail app is trying to access this website.

### What did you expect to happen?
Portmaster should determine the right application that is accessing the internet.

### How did you reproduce it?
It happens from time to time.

### Additional information

**Debug-Info**: https://support.safing.io/privatebin/?2bc95ee4098f8f45#762cS8Am9ajtTDUwkuAeTFbAqxEV2BC7Hiz1z88fZhum

**Comment by dhaavi** [comment]
Hey @sedrubal, thanks for reporting this issue.

This normally should not happen.

Can you send us the detailed connection info from a connection from Firefox and one from the mail app?
Just click "Copy JSON" with the details of a connection open. (You might need to switch to the Advanced Interface in the top right).
Please choose connections that you don't mind sharing, as the information posted here is public.

**Comment by tukusejssirs** [comment]
Recently, I began to notice that the process names (their association) is worse that it used to be (e.g. a `docker pull` in a terminal shows me a prompt for Mailspring app). My guess is that the incorrect association is close to 50 %, maybe a bit less. I used to be better before, at least it felt so (subjectively).

Note that I am using version `1.0.0` on beta release channel. I use Arch Linux and Linux kernel version `linux@6.0.6.arch1-1` if that might be the source of an issue.
